src/server/utilities/liteATIClickTracking/index.ts

```diff
@@ -0,0 +1,93 @@
+export default () => {
+  window.addEventListener('load', () => {
+    document.addEventListener('click', (event: MouseEvent) => {
+      const targetElement = event.target as HTMLElement;
+      // eslint-disable-next-line no-undef
+      if (targetElement?.tagName === 'A') {
+        event.stopPropagation();
+        event.preventDefault();
+
+        const atiURL = targetElement.getAttribute('data-lite-ati-tracking');
+
+        if (atiURL == null) {
+          return;
+        }
+
+        const currentAnchorElement = event.currentTarget as HTMLAnchorElement;
+        const nextPageUrl = currentAnchorElement?.href;
+
+        const {
+          screen: { width, height, colorDepth, pixelDepth },
+          innerWidth,
+          innerHeight,
+        } = window;
+        const now = new Date();
+        const hours = now.getHours();
+        const mins = now.getMinutes();
+        const secs = now.getSeconds();
+
+        // COOKIE SETTINGS
+        const cookieName = 'atuserid';
+        const expires = 397; // expires in 13 months
+        const cookiesForPage = `; ${document.cookie}`;
+        const atUserIdCookie = cookiesForPage.split(`; ${cookieName}=`);
+        let atUserIdValue = null;
+
+        if (atUserIdCookie.length === 2) {
+          const cookieInfo = atUserIdCookie.pop()?.split(';').shift();
+
+          if (cookieInfo) {
+            const decodedCookie = decodeURI(cookieInfo);
+            const user = JSON.parse(decodedCookie);
+            atUserIdValue = user.val;
+          }
+        }
+
+        if (!atUserIdValue && crypto.randomUUID) {
+          atUserIdValue = crypto.randomUUID();
+        }
+
+        const stringifiedCookieValue = JSON.stringify({ val: atUserIdValue });
+        if (atUserIdValue) {
+          document.cookie = `${cookieName}=${stringifiedCookieValue}; path=/; max-age=${expires}; Secure;`;
+        }
+
+        const rValue = [
+          width || 0,
+          height || 0,
+          colorDepth || 0,
+          pixelDepth || 0,
+        ].join('x');
+
+        const reValue = [innerWidth || 0, innerHeight || 0].join('x');
+
+        const hlValue = [hours, mins, secs].join('x');
+
+        let clientSideAtiURL = atiURL
+          .concat('&', 'r=', rValue)
+          .concat('&', 're=', reValue)
+          .concat('&', 'hl=', hlValue);
```

Comment:
Can we use URLSearchParams here? It should be supported on all browsers..? Although we may need to check: https://caniuse.com/?search=URLSearchParams

```suggestion
        const params = new URLSearchParams();
        params.append('r', screenResolutionColourDepth);
        params.append('re', browserViewportResolution);
        params.append('hl', timestamp);
```

If not, could we do something which is a bit more readable - using object key values before appending them all at the bottom - this will make it a bit more future proof if we ever decide to add new params. 

```suggestion
        const params = { 
                r: screenResolutionColourDepth, 
                re: browserViewportResolution,
                 ...
                ...(navigator.language && { lng: navigator.language })
        }
```

Reply:
We tried this but found out that 'Object.entries' is not supported by Opera Mini - (then checked CanIUse - https://caniuse.com/?search=Object.entries)

Did find this suggested polyfill, but it looks harder to maintain - https://stackoverflow.com/questions/45849831/object-entries-alternative-for-internet-explorer-and-reactjs

Reply:
Or how about an old-fashioned `for` loop? Or `Object.keys` which looks like it is supported?

Reply:
Yes! Object.keys does work. Oops overlooked that one. This is good practice in writing Vanilla JS 😅